_posts/2021-04-26-changelog-146.markdown

```diff
@@ -0,0 +1,187 @@
+
+* Improve experimental crate-local index:
+
+  * [#6801] Add `Cargo.toml` inspection that checks crate dependency version requirements that do not match any known crate version (by [@Kobzol])
+
+  * [#6599] Use semantic versioning to improve version sorting in dependency completion
+
+  Note, the corresponding feature is disabled by default for now. To use it, enable
+  `org.rust.crates.local.index` [experimental feature][experimental features]
+
+* [#7142] Drop support of `org.rust.cargo.fetch.out.dir` experimental feature.
+  Use `org.rust.cargo.evaluate.build.scripts` instead if you need to evaluate
+  `env!("OUT_DIR")` in your `include!` macro calls
+
+## Fixes
+
+* Name resolution fixes:
+
+  * [#7119] Correctly resolve qualified [attribute](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) procedural macro paths (like `#[foo::bar]`)
+
+  * [#7132] Resolve [field shorthand](https://doc.rust-lang.org/edition-guide/rust-2018/data-types/field-init-shorthand.html) to a local variable in the case of unresolved struct literal
+
+* Improvements of [new name resultion](https://github.com/intellij-rust/intellij-rust/issues/6217) engine:
+
+  * [#7089] Correctly resolve macros with absolute paths (like `::foo::bar!()`)
+
+  * [#7057] Exclude some cfg-disabled items from completion
+
+* `Change Signature` refactoring (`Refactor | Change Signature` or <kbd>Ctrl</kbd>+<kbd>F6</kbd>) fixes:
+
+  * [#7081] Fix some false positives in name conflict detection (by [@Kobzol])
+
+  * [#7080] Do not needlessly import default type arguments (by [@Kobzol])
+
+  * [#7076] Correctly place function visibility after attributes (by [@Kobzol])
+
+* [#7108] Fix auto-import: insert correct import in the case of a renamed cargo package
+
+* [#7101] Fix false positive in `Needless lifetimes` inspection when the lifetime is used in `where` clause (by [@Kobzol])
+
+* [#5491] Improve built-in formatter: add line break after opening brace in struct blocks (by [@Kobzol])
+
+* [#7131] Fix false-positive proofreading warnings for sentences written in several sequential Rust comments.
+  Note, you can enable/disable proofreading in Rust files in `Preferences | Editor | Proofreading | Grammar` settings
+
+* [#7121] Reload project model automatically when workspace-root `Cargo.toml` file is changed (and saved).
+  Previously it was work only for package `Cargo.toml` files
+
+* [#7079] Fix proc macro expansion if proc macro crate is added as separate cargo project in `Cargo` tool window
+
+* [#7072] Place `Show the result of macro expansion` intention lower in the intention list
+
+* [#6013] Don't run some inspections on Rust files outside a valid Cargo project to avoid expected false positive errors
+
+* [#7070] Don't show errors related to Rust edition if we don't known the edition (by [@Stzx])
+
+* [#7004] Fix experimental crate-local index: don't perform code completion and run inspections if the index is not ready
```

Comment:
We can move this item to feature section in `Improve experimental crate-local index` group. I know that it's not a feature but I think it's ok
If it's ok for you, let also adjust a text like
```suggestion
* [#7004] Don't perform code completion and run inspections if the index is not ready
```